akka-stream/src/main/scala/akka/stream/impl/Throttle.scala

```diff
@@ -20,21 +35,32 @@ private[stream] class Throttle[T](cost: Int,
                                   mode: ThrottleMode)
   extends SimpleLinearGraphStage[T] {
   require(cost > 0, "cost must be > 0")
-  require(per.toMillis > 0, "per time must be > 0")
+  require(per.toNanos > 0, "per time must be > 0")
   require(!(mode == ThrottleMode.Enforcing && maximumBurst < 0), "maximumBurst must be > 0 in Enforcing mode")
+  require(per.toNanos >= cost, "Rates larger than 1 unit / nanosecond are not supported")
+
+  // There is some loss of precision here because of rounding, but this only happens if nanosBetweenTokens is very
+  // small which is usually at rates where that precision is highly unlikely anyway as the overhead of this stage
+  // is likely higher than the required accuracy interval.
+  private val nanosBetweenTokens = (per.toNanos.toDouble / cost).toLong
```

Comment:
I don’t see how `.toDouble` would have an effect here (apart from introducing rounding errors for weirdly specified rates).